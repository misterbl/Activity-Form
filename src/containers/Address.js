import React from "react";
import { Formik } from "formik";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";
import Modal from "react-modal";
import * as Yup from "yup";
import * as apiThunk from "../actions/thunks/apiThunk";
import AddressForm from "../components/AddressForm";
import { getRegisteredAddresses } from "../selectors/apiSelectors";
import { saveActivityAddress } from "../actions/actionCreators/apiActions";
import ROUTES from "../const/route";
import ExistingAddressForm from "../components/ExistingAddressForm";

export class Address extends React.Component {
  state = {
    modalIsOpen: false,
    selectedAddress: {
      buildingUnit: "",
      buildingName: "",
      streetNumber: "",
      streetName: "",
      town: ""
    }
  };

  openModal = () => {
    this.setState({ modalIsOpen: true });
  };

  closeModal = () => {
    this.setState({ modalIsOpen: false });
  };

  handleSubmit = data => {
    const {
      saveActivityAddress,
      history: { push }
    } = this.props;
    saveActivityAddress(data);
    push(ROUTES.CONFIRMATION);
  };

  backToPreviousPage = () => {
    this.props.history.push(ROUTES.ACTIVITY);
  };

  onClick = async () => {
    await this.props.getAddresses();
    this.openModal();
  };

  onAddressSelect = event => {
    this.setState({
      selectedAddress: this.props.addresses[Number(event.selectAddress)],
      modalIsOpen: false
    });
  };

  render() {
    const { addresses } = this.props;
    const { selectedAddress } = this.state;
    return (
      <main className="bg-white form-container px-5 pb-5 pt-4">
        <div className="d-flex justify-content-center">
          <h1 className="text-center title-30 ml-5 mb-5 w-100 pt4">
            Add the address
          </h1>
          <button
            type="button"
            id="copy-activity-button"
            className="btn copy-activity-button ml-3"
            onClick={this.onClick}
          >
            Copy from Existing Activity
          </button>
          <Modal
            appElement={document.getElementById("root")}
            isOpen={this.state.modalIsOpen}
            onRequestClose={this.closeModal}
            contentLabel="Address selection"
          >
            <button className="close-modal-button " onClick={this.closeModal}>
              X
            </button>
            {addresses && <div>Hello</div>}
            {addresses && (
              <div className="mt-5">
                <Formik
                  initialValues={{
                    selectAddress: ""
                  }}
                  onSubmit={this.onAddressSelect}
                  render={formikProps => (
                    <ExistingAddressForm
                      {...formikProps}
                      addresses={addresses}
                    />
                  )}
                />
              </div>
            )}
          </Modal>
        </div>
        <Formik
          enableReinitialize
          validationSchema={Yup.object().shape({
            streetNumber: Yup.string().required("Street number requiered"),
            streetName: Yup.string().required("Street name requiered"),
            town: Yup.string().required("Town requiered"),
            postcode: Yup.string().required("Postcode requiered")
          })}
          initialValues={{
            buildingUnit: selectedAddress.buildingUnit,
            buildingName: selectedAddress.buildingName,
            streetNumber: selectedAddress.streetNumber,
            streetName: selectedAddress.streetName,
            town: selectedAddress.town,
            postcode: selectedAddress.postcode
          }}
          onSubmit={this.handleSubmit}
          render={formikProps => (
            <AddressForm
              {...formikProps}
              backToPreviousPage={this.backToPreviousPage}
            />
          )}
        />
      </main>
    );
  }
}

export const mapStateToProps = state => ({
  addresses: getRegisteredAddresses(state)
});

export const mapDispatchToProps = {
  getAddresses: apiThunk.getAddresses,
  saveActivityAddress
};

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(Address)
);
