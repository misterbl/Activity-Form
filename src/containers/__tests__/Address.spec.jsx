import * as React from "react";
import { shallow } from "enzyme";
import historyMock from "../../testMocks/history.mock";
import { Address } from "../Address";
import ROUTES from "../../const/route";
import formikRenderMock from "../../testMocks/formikRender.mock";

describe("Address", () => {
  const props = {
    values: {
      streetNumber: "streetNumberTest",
      streetName: "streetNameTest",
      town: "townTest",
      postcode: "postcodeTest"
    },
    errors: {
      streetNumber: null,
      streetName: null,
      town: null,
      postcode: null
    },
    ...formikRenderMock,
    initialValues: {
      streetNumber: "",
      streetName: "",
      town: "",
      postcode: ""
    },
    history: historyMock
  };
  const wrapper = shallow(<Address {...props} />);
  it("matches the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render a Formik element", () => {
    expect(wrapper.find("Formik").length).toBe(1);
  });
  describe("backToPreviousPage", () => {
    it("calls history.push", () => {
      wrapper.instance().backToPreviousPage();
      expect(props.history.push).toHaveBeenCalledWith(ROUTES.ACTIVITY);
    });
  });
});
