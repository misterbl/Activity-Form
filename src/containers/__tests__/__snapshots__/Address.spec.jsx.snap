// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Address matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Address
    dirty={false}
    errors={Object {}}
    handleBlur={[Function]}
    handleChange={[Function]}
    handleReset={[Function]}
    handleSubmit={[Function]}
    history={
      Object {
        "action": "PUSH",
        "block": [MockFunction],
        "createHref": [MockFunction],
        "go": [MockFunction],
        "goBack": [MockFunction],
        "goForward": [MockFunction],
        "length": 123,
        "listen": [MockFunction],
        "location": Object {
          "hash": "",
          "key": "",
          "pathname": "",
          "search": "",
          "state": "",
        },
        "push": [MockFunction],
        "replace": [MockFunction],
      }
    }
    initialValues={
      Object {
        "postcode": "",
        "streetName": "",
        "streetNumber": "",
        "town": "",
      }
    }
    isSubmitting={false}
    isValid={false}
    isValidating={false}
    registerField={[Function]}
    resetForm={[Function]}
    setError={[Function]}
    setErrors={[Function]}
    setFieldError={[Function]}
    setFieldTouched={[Function]}
    setFieldValue={[Function]}
    setFormikState={[Function]}
    setStatus={[Function]}
    setSubmitting={[Function]}
    setTouched={[Function]}
    setValues={[Function]}
    submitCount={0}
    submitForm={[Function]}
    touched={Object {}}
    unregisterField={[Function]}
    validateField={[Function]}
    validateForm={[Function]}
    values={
      Object {
        "postcode": "postcodeTest",
        "streetName": "streetNameTest",
        "streetNumber": "streetNumberTest",
        "town": "townTest",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="d-flex justify-content-center"
        >
          <h1
            className="text-center font-30 ml-5 mb-5 w-100 pt4"
          >
            Add the address
          </h1>
          <button
            className="btn copy-activity-button ml-3"
            id="copy-activity-button"
            onClick={[Function]}
            type="button"
          >
            Copy from Existing Activity
          </button>
          <Modal
            appElement={null}
            ariaHideApp={true}
            bodyOpenClassName="ReactModal__Body--open"
            closeTimeoutMS={0}
            contentLabel="Address selection"
            isOpen={false}
            onRequestClose={[Function]}
            parentSelector={[Function]}
            portalClassName="ReactModalPortal"
            role="dialog"
            shouldCloseOnEsc={true}
            shouldCloseOnOverlayClick={true}
            shouldFocusAfterRender={true}
            shouldReturnFocusAfterClose={true}
          >
            <button
              className="close-modal-button "
              onClick={[Function]}
            >
              X
            </button>
          </Modal>
        </div>,
        <Formik
          enableReinitialize={true}
          initialValues={
            Object {
              "buildingName": "",
              "buildingUnit": "",
              "postcode": "",
              "streetName": "",
              "streetNumber": "",
              "town": "",
            }
          }
          isInitialValid={false}
          onSubmit={[Function]}
          render={[Function]}
          validateOnBlur={true}
          validateOnChange={true}
          validationSchema={
            ObjectSchema {
              "_blacklist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "_conditions": Array [],
              "_defaultDefault": [Function],
              "_deps": Array [],
              "_excludedEdges": Array [],
              "_exclusive": Object {},
              "_mutate": false,
              "_nodes": Array [
                "postcode",
                "town",
                "streetName",
                "streetNumber",
              ],
              "_options": Object {
                "abortEarly": true,
                "recursive": true,
              },
              "_type": "object",
              "_typeError": [Function],
              "_whitelist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "fields": Object {
                "postcode": StringSchema {
                  "_blacklist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "_conditions": Array [],
                  "_deps": Array [],
                  "_exclusive": Object {
                    "required": false,
                    "undefined": false,
                  },
                  "_mutate": false,
                  "_options": Object {
                    "abortEarly": true,
                    "recursive": true,
                  },
                  "_type": "string",
                  "_typeError": [Function],
                  "_whitelist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "tests": Array [
                    [Function],
                    [Function],
                    [Function],
                  ],
                  "transforms": Array [
                    [Function],
                  ],
                },
                "streetName": StringSchema {
                  "_blacklist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "_conditions": Array [],
                  "_deps": Array [],
                  "_exclusive": Object {
                    "required": false,
                  },
                  "_mutate": false,
                  "_options": Object {
                    "abortEarly": true,
                    "recursive": true,
                  },
                  "_type": "string",
                  "_typeError": [Function],
                  "_whitelist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "tests": Array [
                    [Function],
                    [Function],
                  ],
                  "transforms": Array [
                    [Function],
                  ],
                },
                "streetNumber": StringSchema {
                  "_blacklist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "_conditions": Array [],
                  "_deps": Array [],
                  "_exclusive": Object {
                    "required": false,
                  },
                  "_mutate": false,
                  "_options": Object {
                    "abortEarly": true,
                    "recursive": true,
                  },
                  "_type": "string",
                  "_typeError": [Function],
                  "_whitelist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "tests": Array [
                    [Function],
                    [Function],
                  ],
                  "transforms": Array [
                    [Function],
                  ],
                },
                "town": StringSchema {
                  "_blacklist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "_conditions": Array [],
                  "_deps": Array [],
                  "_exclusive": Object {
                    "required": false,
                  },
                  "_mutate": false,
                  "_options": Object {
                    "abortEarly": true,
                    "recursive": true,
                  },
                  "_type": "string",
                  "_typeError": [Function],
                  "_whitelist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "tests": Array [
                    [Function],
                    [Function],
                  ],
                  "transforms": Array [
                    [Function],
                  ],
                },
              },
              "tests": Array [],
              "transforms": Array [
                [Function],
              ],
            }
          }
        />,
      ],
      "className": "bg-white form-container px-5 pb-5 pt-4",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="text-center font-30 ml-5 mb-5 w-100 pt4"
            >
              Add the address
            </h1>,
            <button
              className="btn copy-activity-button ml-3"
              id="copy-activity-button"
              onClick={[Function]}
              type="button"
            >
              Copy from Existing Activity
            </button>,
            <Modal
              appElement={null}
              ariaHideApp={true}
              bodyOpenClassName="ReactModal__Body--open"
              closeTimeoutMS={0}
              contentLabel="Address selection"
              isOpen={false}
              onRequestClose={[Function]}
              parentSelector={[Function]}
              portalClassName="ReactModalPortal"
              role="dialog"
              shouldCloseOnEsc={true}
              shouldCloseOnOverlayClick={true}
              shouldFocusAfterRender={true}
              shouldReturnFocusAfterClose={true}
            >
              <button
                className="close-modal-button "
                onClick={[Function]}
              >
                X
              </button>
            </Modal>,
          ],
          "className": "d-flex justify-content-center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Add the address",
              "className": "text-center font-30 ml-5 mb-5 w-100 pt4",
            },
            "ref": null,
            "rendered": "Add the address",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Copy from Existing Activity",
              "className": "btn copy-activity-button ml-3",
              "id": "copy-activity-button",
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": "Copy from Existing Activity",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "appElement": null,
              "ariaHideApp": true,
              "bodyOpenClassName": "ReactModal__Body--open",
              "children": Array [
                <button
                  className="close-modal-button "
                  onClick={[Function]}
                >
                  X
                </button>,
                undefined,
              ],
              "closeTimeoutMS": 0,
              "contentLabel": "Address selection",
              "isOpen": false,
              "onRequestClose": [Function],
              "parentSelector": [Function],
              "portalClassName": "ReactModalPortal",
              "role": "dialog",
              "shouldCloseOnEsc": true,
              "shouldCloseOnOverlayClick": true,
              "shouldFocusAfterRender": true,
              "shouldReturnFocusAfterClose": true,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "X",
                  "className": "close-modal-button ",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "X",
                "type": "button",
              },
              undefined,
            ],
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "enableReinitialize": true,
          "initialValues": Object {
            "buildingName": "",
            "buildingUnit": "",
            "postcode": "",
            "streetName": "",
            "streetNumber": "",
            "town": "",
          },
          "isInitialValid": false,
          "onSubmit": [Function],
          "render": [Function],
          "validateOnBlur": true,
          "validateOnChange": true,
          "validationSchema": ObjectSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_defaultDefault": [Function],
            "_deps": Array [],
            "_excludedEdges": Array [],
            "_exclusive": Object {},
            "_mutate": false,
            "_nodes": Array [
              "postcode",
              "town",
              "streetName",
              "streetNumber",
            ],
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "object",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "fields": Object {
              "postcode": StringSchema {
                "_blacklist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "_conditions": Array [],
                "_deps": Array [],
                "_exclusive": Object {
                  "required": false,
                  "undefined": false,
                },
                "_mutate": false,
                "_options": Object {
                  "abortEarly": true,
                  "recursive": true,
                },
                "_type": "string",
                "_typeError": [Function],
                "_whitelist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "tests": Array [
                  [Function],
                  [Function],
                  [Function],
                ],
                "transforms": Array [
                  [Function],
                ],
              },
              "streetName": StringSchema {
                "_blacklist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "_conditions": Array [],
                "_deps": Array [],
                "_exclusive": Object {
                  "required": false,
                },
                "_mutate": false,
                "_options": Object {
                  "abortEarly": true,
                  "recursive": true,
                },
                "_type": "string",
                "_typeError": [Function],
                "_whitelist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "tests": Array [
                  [Function],
                  [Function],
                ],
                "transforms": Array [
                  [Function],
                ],
              },
              "streetNumber": StringSchema {
                "_blacklist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "_conditions": Array [],
                "_deps": Array [],
                "_exclusive": Object {
                  "required": false,
                },
                "_mutate": false,
                "_options": Object {
                  "abortEarly": true,
                  "recursive": true,
                },
                "_type": "string",
                "_typeError": [Function],
                "_whitelist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "tests": Array [
                  [Function],
                  [Function],
                ],
                "transforms": Array [
                  [Function],
                ],
              },
              "town": StringSchema {
                "_blacklist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "_conditions": Array [],
                "_deps": Array [],
                "_exclusive": Object {
                  "required": false,
                },
                "_mutate": false,
                "_options": Object {
                  "abortEarly": true,
                  "recursive": true,
                },
                "_type": "string",
                "_typeError": [Function],
                "_whitelist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "tests": Array [
                  [Function],
                  [Function],
                ],
                "transforms": Array [
                  [Function],
                ],
              },
            },
            "tests": Array [],
            "transforms": Array [
              [Function],
            ],
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="d-flex justify-content-center"
          >
            <h1
              className="text-center font-30 ml-5 mb-5 w-100 pt4"
            >
              Add the address
            </h1>
            <button
              className="btn copy-activity-button ml-3"
              id="copy-activity-button"
              onClick={[Function]}
              type="button"
            >
              Copy from Existing Activity
            </button>
            <Modal
              appElement={null}
              ariaHideApp={true}
              bodyOpenClassName="ReactModal__Body--open"
              closeTimeoutMS={0}
              contentLabel="Address selection"
              isOpen={false}
              onRequestClose={[Function]}
              parentSelector={[Function]}
              portalClassName="ReactModalPortal"
              role="dialog"
              shouldCloseOnEsc={true}
              shouldCloseOnOverlayClick={true}
              shouldFocusAfterRender={true}
              shouldReturnFocusAfterClose={true}
            >
              <button
                className="close-modal-button "
                onClick={[Function]}
              >
                X
              </button>
            </Modal>
          </div>,
          <Formik
            enableReinitialize={true}
            initialValues={
              Object {
                "buildingName": "",
                "buildingUnit": "",
                "postcode": "",
                "streetName": "",
                "streetNumber": "",
                "town": "",
              }
            }
            isInitialValid={false}
            onSubmit={[Function]}
            render={[Function]}
            validateOnBlur={true}
            validateOnChange={true}
            validationSchema={
              ObjectSchema {
                "_blacklist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "_conditions": Array [],
                "_defaultDefault": [Function],
                "_deps": Array [],
                "_excludedEdges": Array [],
                "_exclusive": Object {},
                "_mutate": false,
                "_nodes": Array [
                  "postcode",
                  "town",
                  "streetName",
                  "streetNumber",
                ],
                "_options": Object {
                  "abortEarly": true,
                  "recursive": true,
                },
                "_type": "object",
                "_typeError": [Function],
                "_whitelist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "fields": Object {
                  "postcode": StringSchema {
                    "_blacklist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "_conditions": Array [],
                    "_deps": Array [],
                    "_exclusive": Object {
                      "required": false,
                      "undefined": false,
                    },
                    "_mutate": false,
                    "_options": Object {
                      "abortEarly": true,
                      "recursive": true,
                    },
                    "_type": "string",
                    "_typeError": [Function],
                    "_whitelist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "tests": Array [
                      [Function],
                      [Function],
                      [Function],
                    ],
                    "transforms": Array [
                      [Function],
                    ],
                  },
                  "streetName": StringSchema {
                    "_blacklist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "_conditions": Array [],
                    "_deps": Array [],
                    "_exclusive": Object {
                      "required": false,
                    },
                    "_mutate": false,
                    "_options": Object {
                      "abortEarly": true,
                      "recursive": true,
                    },
                    "_type": "string",
                    "_typeError": [Function],
                    "_whitelist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "tests": Array [
                      [Function],
                      [Function],
                    ],
                    "transforms": Array [
                      [Function],
                    ],
                  },
                  "streetNumber": StringSchema {
                    "_blacklist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "_conditions": Array [],
                    "_deps": Array [],
                    "_exclusive": Object {
                      "required": false,
                    },
                    "_mutate": false,
                    "_options": Object {
                      "abortEarly": true,
                      "recursive": true,
                    },
                    "_type": "string",
                    "_typeError": [Function],
                    "_whitelist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "tests": Array [
                      [Function],
                      [Function],
                    ],
                    "transforms": Array [
                      [Function],
                    ],
                  },
                  "town": StringSchema {
                    "_blacklist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "_conditions": Array [],
                    "_deps": Array [],
                    "_exclusive": Object {
                      "required": false,
                    },
                    "_mutate": false,
                    "_options": Object {
                      "abortEarly": true,
                      "recursive": true,
                    },
                    "_type": "string",
                    "_typeError": [Function],
                    "_whitelist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "tests": Array [
                      [Function],
                      [Function],
                    ],
                    "transforms": Array [
                      [Function],
                    ],
                  },
                },
                "tests": Array [],
                "transforms": Array [
                  [Function],
                ],
              }
            }
          />,
        ],
        "className": "bg-white form-container px-5 pb-5 pt-4",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="text-center font-30 ml-5 mb-5 w-100 pt4"
              >
                Add the address
              </h1>,
              <button
                className="btn copy-activity-button ml-3"
                id="copy-activity-button"
                onClick={[Function]}
                type="button"
              >
                Copy from Existing Activity
              </button>,
              <Modal
                appElement={null}
                ariaHideApp={true}
                bodyOpenClassName="ReactModal__Body--open"
                closeTimeoutMS={0}
                contentLabel="Address selection"
                isOpen={false}
                onRequestClose={[Function]}
                parentSelector={[Function]}
                portalClassName="ReactModalPortal"
                role="dialog"
                shouldCloseOnEsc={true}
                shouldCloseOnOverlayClick={true}
                shouldFocusAfterRender={true}
                shouldReturnFocusAfterClose={true}
              >
                <button
                  className="close-modal-button "
                  onClick={[Function]}
                >
                  X
                </button>
              </Modal>,
            ],
            "className": "d-flex justify-content-center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Add the address",
                "className": "text-center font-30 ml-5 mb-5 w-100 pt4",
              },
              "ref": null,
              "rendered": "Add the address",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Copy from Existing Activity",
                "className": "btn copy-activity-button ml-3",
                "id": "copy-activity-button",
                "onClick": [Function],
                "type": "button",
              },
              "ref": null,
              "rendered": "Copy from Existing Activity",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "appElement": null,
                "ariaHideApp": true,
                "bodyOpenClassName": "ReactModal__Body--open",
                "children": Array [
                  <button
                    className="close-modal-button "
                    onClick={[Function]}
                  >
                    X
                  </button>,
                  undefined,
                ],
                "closeTimeoutMS": 0,
                "contentLabel": "Address selection",
                "isOpen": false,
                "onRequestClose": [Function],
                "parentSelector": [Function],
                "portalClassName": "ReactModalPortal",
                "role": "dialog",
                "shouldCloseOnEsc": true,
                "shouldCloseOnOverlayClick": true,
                "shouldFocusAfterRender": true,
                "shouldReturnFocusAfterClose": true,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "X",
                    "className": "close-modal-button ",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "X",
                  "type": "button",
                },
                undefined,
              ],
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "enableReinitialize": true,
            "initialValues": Object {
              "buildingName": "",
              "buildingUnit": "",
              "postcode": "",
              "streetName": "",
              "streetNumber": "",
              "town": "",
            },
            "isInitialValid": false,
            "onSubmit": [Function],
            "render": [Function],
            "validateOnBlur": true,
            "validateOnChange": true,
            "validationSchema": ObjectSchema {
              "_blacklist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "_conditions": Array [],
              "_defaultDefault": [Function],
              "_deps": Array [],
              "_excludedEdges": Array [],
              "_exclusive": Object {},
              "_mutate": false,
              "_nodes": Array [
                "postcode",
                "town",
                "streetName",
                "streetNumber",
              ],
              "_options": Object {
                "abortEarly": true,
                "recursive": true,
              },
              "_type": "object",
              "_typeError": [Function],
              "_whitelist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "fields": Object {
                "postcode": StringSchema {
                  "_blacklist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "_conditions": Array [],
                  "_deps": Array [],
                  "_exclusive": Object {
                    "required": false,
                    "undefined": false,
                  },
                  "_mutate": false,
                  "_options": Object {
                    "abortEarly": true,
                    "recursive": true,
                  },
                  "_type": "string",
                  "_typeError": [Function],
                  "_whitelist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "tests": Array [
                    [Function],
                    [Function],
                    [Function],
                  ],
                  "transforms": Array [
                    [Function],
                  ],
                },
                "streetName": StringSchema {
                  "_blacklist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "_conditions": Array [],
                  "_deps": Array [],
                  "_exclusive": Object {
                    "required": false,
                  },
                  "_mutate": false,
                  "_options": Object {
                    "abortEarly": true,
                    "recursive": true,
                  },
                  "_type": "string",
                  "_typeError": [Function],
                  "_whitelist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "tests": Array [
                    [Function],
                    [Function],
                  ],
                  "transforms": Array [
                    [Function],
                  ],
                },
                "streetNumber": StringSchema {
                  "_blacklist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "_conditions": Array [],
                  "_deps": Array [],
                  "_exclusive": Object {
                    "required": false,
                  },
                  "_mutate": false,
                  "_options": Object {
                    "abortEarly": true,
                    "recursive": true,
                  },
                  "_type": "string",
                  "_typeError": [Function],
                  "_whitelist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "tests": Array [
                    [Function],
                    [Function],
                  ],
                  "transforms": Array [
                    [Function],
                  ],
                },
                "town": StringSchema {
                  "_blacklist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "_conditions": Array [],
                  "_deps": Array [],
                  "_exclusive": Object {
                    "required": false,
                  },
                  "_mutate": false,
                  "_options": Object {
                    "abortEarly": true,
                    "recursive": true,
                  },
                  "_type": "string",
                  "_typeError": [Function],
                  "_whitelist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "tests": Array [
                    [Function],
                    [Function],
                  ],
                  "transforms": Array [
                    [Function],
                  ],
                },
              },
              "tests": Array [],
              "transforms": Array [
                [Function],
              ],
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
